### PR TITLE
EventSystem (New crew member, pirates, etc)

### PR DESCRIPTION
**Pull request by relinu**
Adds EventSystem, events can be created in the StreamingAssets/GameEvents folder as xml code. (e.g. attacks from space pirates, meteors, a new crew member,  merchants)

### CODE REVIEW

File: Assets/Scripts/Controllers/GameEventManager.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            CreateEvent(name, functionNames.ToArray());
+        }
+    }
+
+    void CreateEvent(string eventName, string[] functionNames)
+    {
+        GameEvent gameEvent = new GameEvent(eventName);
+
+        foreach(string funcName in functionNames){
+            gameEvent.RegisterExecutionAction(funcName);
```
Comment (NogginBops) [inline Assets/Scripts/Controllers/GameEventManager.cs]: Maybe a AddAll method?

**Comment by bjubes** [comment]
How are events present to the player? Is their a corresponding pop up that warns the player kinda like rimworld? Or do they just happen in real time without warning?

**Comment by relinu** [comment]
This can be set in the lua code in the future maybe there you can call a method that display a mesaage on text or changes something in realtime like adding a pirate at the dge of the world that runs to your base and destroys it (like in rimworld)

**Comment by relinu** [comment]
Lua code is added

**Comment by relinu** [comment]
screenshot06

Changed event to spawn a new character every 30 seconds... Quiet funny with that many little guys :D

**Comment by henninglive** [comment]
The spawn a new character event does kind of conflict with #253, which also way of adding new characters.

**Comment by quill18** [comment]
Merged!

**Comment by ComputerOverlord** [comment]
@quill18 I would have used #253 But, this is fine.

**Comment by relinu** [comment]
@ComputerOverlord we can use that too, the current event is only a example of how one can create a event and will definitely be changed in the future

**Comment by henninglive** [comment]
I would prefer we remove the spawn a new character event sooner rather than later. The game begins to lag quite quickly as they multiply.

**Comment by MANDAL0R3** [comment]
This is already merged but might still be a nice place to discuss this. The cloning pod I created might not indefinitely be used to create new crew members. As of right now im leaving it as it is because it works and adds a player controlled way to add new crew members. I have been thinking that instead of creating new crew members, it instead is used as a one time respawn that has to be assigned to a character before being damaged and inoperable.

**Comment by vogonistic** [comment]
@MANDAL0R3 There is an issue open to discuss how characters/crew members are supposed to be created #283. Can you open a PR that fixes the error in `Character.cs`?

**Pull request merged**